pulsar-transaction/coordinator/src/main/java/org/apache/pulsar/transaction/coordinator/impl/TxnBatchedPositionImpl.java

```diff
@@ -56,4 +61,19 @@ public boolean equals(Object o) {
     public int hashCode() {
         return Objects.hash(super.hashCode(), batchSize, batchIndex);
     }
+
+    /**
+     * Build the attribute ackSet to that {@link #batchIndex} is false and others is true.
+     */
+    public void deleteFromAckSet(){
```

Comment:
```suggestion
    void setAckSetByIndex()
```

Reply:
And please add a test to cover the new method.

Reply:
Good idea. A bug was found by adding this unit tests.

Reply:
Already fixed